scripts/test/Muon/PlottingPresenter_test.py

```diff
@@ -3,29 +3,77 @@
 import os
 os.environ["QT_API"] = "pyqt"  # noqa E402
 
+from Muon.GUI.ElementalAnalysis.Plotting.edit_windows.select_subplot import SelectSubplot
 from Muon.GUI.ElementalAnalysis.Plotting.plotting_presenter import PlotPresenter
 from Muon.GUI.ElementalAnalysis.Plotting.plotting_view import PlotView
+from Muon.GUI.ElementalAnalysis.Plotting.edit_windows.remove_plot_window import RemovePlotWindowView
+from  Muon.GUI.Common import mock_widget
 
+from qtpy import QtCore
 
 try:
     from unittest import mock
 except ImportError:
     import mock
 
+# simple class to mock QDialogs
+class dummy_popup(object):
+    def __init__(self,lines,parent=None):
+        self.dummy = True
+
+    def setMinimumSize(self, min1,min2):
+        return 
+
+    def show(self):
+        return
+
+    def raise_(self):
+        return
 
 class PlottingPresenterTest(unittest.TestCase):
     def setUp(self):
         view = mock.create_autospec(PlotView)
+        view.subplotRemovedSignal = mock.Mock()
         self.presenter = PlotPresenter(view)
         self.presenter.view.canvas = mock.Mock()
         self.presenter.view.canvas.draw = mock.Mock()
+
         self.view = self.presenter.view
+        self.view.close = mock.Mock()
+        self.view.get_subplots = mock.Mock(return_value = {})
+        self.view.removeLine = mock.Mock()
+
+
+        # explicitly mock pop up
+        self.mock_selector =  mock.create_autospec(dummy_popup)
+        self.mock_selector.subplotSelectorSignal = mock.Mock()
+        self.mock_selector.closeEventSignal = mock.Mock()
+
+        self.mock_rmWindow =  mock.create_autospec(dummy_popup)
+        self.mock_rmWindow.applyRemoveSignal = mock.Mock()
+        self.mock_rmWindow.closeEventSignal = mock.Mock()
+        self.mock_rmWindow.subplot = mock.Mock(return_value="plot")
+        self.mock_rmWindow.getState = mock.Mock(side_effect = [True, True])
+        self.mock_rmWindow.getLine = mock.Mock()
+
+        # mock presenter to create mock pop ups
+        self.presenter.createRmWindow = mock.Mock(return_value = self.mock_rmWindow)
+        self.presenter.createSelectWindow = mock.Mock(return_value =self.mock_selector)
 
         self.mock_name = mock.Mock()
         self.mock_workspace = mock.Mock()
         self.mock_func = mock.Mock()
         self.mock_arbitrary_args = [mock.Mock() for i in range(3)]
 
+
+    def test_setup(self):
+        assert(self.view.setRmConnection.call_count ==1)
```

Comment:
Why not use the `assert*` methods provided by the `unittest` module in this file?

Reply:
I forgot they existed. I can change them